lib/filters/validator.js

```diff
@@ -43,44 +42,44 @@ const supportedValidators = ['maximum',
  * @constructor
  */
 class Validator {
-    constructor(parameters, definitions) {
+    constructor(parameters = [], requestBody = {}, schemas) {
         this._ajv = constructAjv({ verbose: true });
-        if (definitions) {
-            Object.keys(definitions).forEach((schemaName) => {
+        if (schemas) {
+            Object.keys(schemas).forEach((schemaName) => {
                 this._ajv.addSchema(
-                    definitions[schemaName],
-                    `#/definitions/${schemaName}`
+                    schemas[schemaName],
+                    `#/components/schemas/${schemaName}`
                 );
             });
         }
 
         this._paramCoercionFunc = this._createTypeCoercionFunc(parameters.filter((p) => {
-            return p.in !== 'formData' && p.in !== 'body' && p.type !== 'string';
+            return p.schema.type !== 'string';
         }));
-        const bodyParams = parameters.filter((p) => p.in === 'body');
-        if (bodyParams.length > 1) {
-            throw new Error('Only a single body parameter allowed');
-        } else if (bodyParams.length) {
-            // Have a body parameter, special-case coercion to support formData and JSON
-            const bodyParam = bodyParams[0];
-            if (bodyParam.schema && bodyParam.schema.properties) {
+
+        if (requestBody.content) {
+            // Have a body parameter, special-case coercion to support form-data and JSON
+            const reqContent = requestBody.content;
+            const bodyType = Object.keys(reqContent)[0];
```

Comment:
Currenly we will break if there's more then one possible content-types of the body and not the first one of the list is submitted. We don't use multi-type POST endpoints now, but could you add a comment explaining that?

Reply:
Done.